### PR TITLE
oEmbed for Facebook, Instagram and Twitter

### PR DESCRIPTION
**Pull request by s-tyshchenko**
oEmbed update:

- Support of oEmbed for Twitter, Facebook and Instagram
- Redesigned of oEmbed settings pages
- Added dynamic endpoint parameters inputs

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Issue: https://github.com/humhub/humhub/issues/5494

Screenshots:

Screenshot 2022-01-22 в 00 31 07

Screenshot 2022-01-22 в 00 11 34

### CODE REVIEW

File: protected/humhub/migrations/m220121_193617_oembed_setting_update.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * Class m220121_193617_oembed_setting_update
+ */
+class m220121_193617_oembed_setting_update extends Migration
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function up()
+    {
+        $oembedProvidersJson = Json::encode([
```
Comment (luke-) [inline protected/humhub/migrations/m220121_193617_oembed_setting_update.php:16]: Can we somehow keep here OEmbed providers already defined by users themselves?

File: protected/humhub/modules/admin/views/setting/oembed.php [hunk 2]
```diff
@@ -19,20 +19,27 @@
...
 
 <?php if (count($providers) != 0): ?>
-    <ul>
-        <?php foreach ($providers as $providerUrl => $providerOEmbedAPI) : ?>
-            <li><?= Html::a($providerUrl, Url::to(['oembed-edit', 'prefix' => $providerUrl]), ['data-method' => 'POST']); ?></li>
+    <div id="oembed-providers">
+        <?php foreach ($providers as $providerName => $provider) : ?>
+            <div class="oembed-provider-container col-xs-6 col-md-3">
+                <div class="oembed-provider">
+                    <span><?= $providerName ?></span>
```
Comment (luke-) [inline protected/humhub/modules/admin/views/setting/oembed.php:26]: Can you please add a `Html::encode' here.

**Review by luke- — CHANGES_REQUESTED** [review]
@s-tyshchenko Thanks, please see my comments.

Can you please also adapt the Changelog?
https://docs.humhub.org/docs/develop/pull-requests#changelog

**Review by luke- — APPROVED** [review]

**Comment by luke-** [comment]
@s-tyshchenko Is there a display for providers that need an API key but where none is configured?

**Comment by s-tyshchenko** [comment]
> @s-tyshchenko Is there a display for providers that need an API key but where none is configured?

No, but I can add a parameter to the config indicating that an API key is required for this provider.

**Comment by luke-** [comment]
@s-tyshchenko Could we check if an Access Token variable is used in the Endpoint URL, if yes and the variable is empty, we can indicate that the configuration is incomplete?

File: protected/humhub/modules/admin/views/setting/oembed_edit.php [hunk 3]
```diff
@@ -25,21 +86,30 @@
...
+<?= $form->field($model, 'name')->textInput(['class' => 'form-control']); ?>
+
+<?= $form->field($model, 'pattern')->textInput(['class' => 'form-control']); ?>
+<p class="help-block"><?= Yii::t('AdminModule.settings', 'Regular expression by which the link match will be checked.'); ?></p>
 
 <?= $form->field($model, 'endpoint')->textInput(['class' => 'form-control']); ?>
 <p class="help-block"><?= Yii::t('AdminModule.settings', 'Use %url% as placeholder for URL. Format needs to be JSON. (e.g. http://www.youtube.com/oembed?url=%url%&format=json)'); ?></p>
 
+<?php if(isset($query['access_token'])): ?>
+    <?= $form->field($model, 'access_token')->textInput(['class' => 'form-control endpoint-param', 'data-param-name' => 'access_token', 'value' => $query['access_token']]) ?>
```
Comment (ArchBlood) [inline protected/humhub/modules/admin/views/setting/oembed_edit.php:102]: Would it be possible to add an entry for `'type' => 'password'` to automatically hide the entered text here, or is this overkill @luke-?

**Review by luke- — APPROVED** [review]

File: protected/humhub/modules/admin/views/setting/oembed_edit.php (hunk 3 above)
Comment (luke-) [reply protected/humhub/modules/admin/views/setting/oembed_edit.php:102]: @ArchBlood Thanks, usually it makes sense, but the token is appended plaintext to the URL anyway, so I would ignore it in this case :-)

**Pull request merged**